skimage/feature/scattering.py

```diff
@@ -0,0 +1,286 @@
+    S_tree = {}
+    # Start computing the scattering coefficients
+    # Zero order scattering coeffs
+    S[:, 0, :, :] = _apply_lowpass(x, wavelet_filters['phi'][current_res],
+                                   J,  spatial_coefs)
+    S_tree[0] = S[:, 0, :, :]
+    l_indexing = np.arange(0, L)
+    # First order scattering coeffs
+    if m > 0:
+        Sview = S[:, 1:J*L+1, :, :].view()
+        Sview.shape = (num_signals, J, L, spatial_coefs, spatial_coefs)
+        # precompute the fourier transform of the images
+        X = np.fft.fft2(x)
+        for j in np.arange(J):
+            filtersj = wavelet_filters['psi'][current_res][j].view()
+            # resolution for the next layer
+            resolution = max(j-oversample, 0)
+            v_resolution.append(resolution)
+            # fft2(| x conv Psi_j |): X is full resolution,
+            # as well as the filters
+            aux = _subsample(np.fft.ifft2(_apply_fourier_mult(X, filtersj)),
+                             resolution)
+            V.append(aux)
+            U.append(np.abs(aux))
+            filter = wavelet_filters['phi'][resolution]
+            Sview[:, j, :, :, :] = _apply_lowpass(U[j],
+                                                  filter, J,
+                                                  spatial_coefs)
+            j_indexing = np.ones((L))*int(j)
+            for i_l, first_order_label in \
+                    enumerate(zip(j_indexing, l_indexing)):
+                S_tree[first_order_label] = Sview[:, j, i_l, :, :]
+    # Second order scattering coeffs
+    if m > 1:
+        sec_order_coefs = int(J*(J-1)*L**2/2)
+        S2norder = S[:, J*L+1:num_coefs, :, :]  # view of the data
+        S2norder.shape = (num_signals, int(sec_order_coefs/L),
+                          L, spatial_coefs, spatial_coefs)
+        indx = 0
+        for j1 in np.arange(J):
+            # U is in the spatial domain
+            Uj1 = np.fft.fft2(U[j1].view())
+            current_res = v_resolution[j1]
+
+            for l1 in np.arange(Uj1.shape[1]):
+                # all images single angle, all spatial coefficients
+                Ujl1 = Uj1[:, l1, ].view()
+
+                layer1_indexing = [(int(j1), int(l1))] * L  # for S_tree
+
+                for j2 in np.arange(j1+1, J):
+                    # | U_lambda1 conv Psi_lambda2 | conv phi
+                    filter = wavelet_filters['psi'][current_res][j2]
+                    aux = np.abs(np.fft.ifft2(
+                        _apply_fourier_mult(Ujl1, filter)
+                    ))
+                    # computing all angles at once
+                    S2norder[:, indx, :, :, :] = \
+                        _apply_lowpass(aux,
+                                       wavelet_filters['phi'][current_res],
+                                       J, spatial_coefs)
+
+                    # save tree structure
+                    j2_indexing = np.ones((L)) * int(j2)
```

Comment:
remove `int`, and `(L) -> L`. Maybe need to specify `dtype` of `ones`, maybe not